design/pushed-authorization-requests.md

```diff
@@ -0,0 +1,350 @@
+
+- **expires_in**
+
+A JSON number that represents the lifetime of the request URI in seconds as a positive integer. The request URI lifetime should be configurable (e.g., between 5 and 600 seconds)
+
+#### PAR Error
+
+-   **error** An error code which can take one of the following values :
+
+    -   **invalid_request** The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed.
+    -   **invalid_client** Client authentication failed, due to missing or invalid client credentials.
+    -   **unauthorized_client** The client is not registered for the requested `response_type`.
+    -   **unsupported_response_type** The server doesn't support the requested `response_type`.
+-   **[ error_description ]** Optional text providing additional information about the error that occurred.
+
+-   **[ error_uri ]** Optional URI for a web page with information about the error that occurred.
+
+
+### Authorization Request
+
+The client uses the request URI value to create the subsequent authorization request by directing the user-agent to make an HTTP request to the authorization server's authorization endpoint like the following:
+
+````
+GET /authorize?client_id=s6BhdRkqt3&request_uri=urn%3Aietf%3Aparams
+       %3Aoauth%3Arequest_uri%3Abwc4JK-ESC0w8acc191e-Y1LTC2 HTTP/1.1
+     Host: as.example.com
+````
+
+## Implementation details
+
+### Server Metadata require_pushed_authorization_requests parameter 
+
+when `false`
+
+there is no mandatory PAR request before autorization endpoint, the client may use it or not
+
+when `true`
+
+PAR request is mandatory prior autorization endpoint and reject any authorization request without a request URI issued from the PAR endpoint, clients must use it
+
+Classes/methods affected:
+
+* org.keycloak.protocol.oidc.OIDCConfigAttributes
+* org.keycloak.protocol.oidc.OIDCWellKnownProvider
+    * getConfig()
+
+### Server Metadata request_uri_lifespan configuration
+The default lifetime of request_uri should be 60 seconds
+
+Classes/methods affected:
+
+* org.keycloak.protocol.oidc.OIDCConfigAttributes
+* org.keycloak.protocol.oidc.OIDCWellKnownProvider
+    * getConfig()
+
+### Client Metadata require_pushed_authorization_requests
+
+when `false`, PAR is not mandatory
+
+when `true`, PAR is mandatory
+
+Classes/methods affected:
+
+* org.keycloak.models.ClientModel
```

Comment:
I think it will be good to not change all the client classes just due this configuration option. Configuration can be saved as client attributes IMO. It will be good to introduce another configuration class with server-specific configuration as well as client-specific configuration. For example see CibaConfig or OAuth2DeviceConfig for inspiration

Reply:
Yes, agree.